kyaml/fn/framework/example_filter_GVK_test.go

```diff
@@ -0,0 +1,34 @@
+package framework_test
+
+import (
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+)
+
+// This example implements a function that reads the desired replicas from the
+// functionConfig and updates the replicas field for all deployments.
+
+func Example_filterGVK() {
+	if err := command.AsMain(framework.ResourceListProcessorFunc(updateReplicas)); err != nil {
+		os.Exit(1)
+	}
+}
+
+// updateReplicas sets a field in resources selecting by GVK.
+func updateReplicas(rl *framework.ResourceList) error {
+	if rl.FunctionConfig == nil {
```

Comment:
Nit: what matters is not the function config missing, but rather it having a replicas field specifically. I would suggest making `GetNestedInt` handle the nil case so all function authors don't have to. And then returning the typed error (ideally with the required path in it!) at L26 instead.